analytics/views.py

```diff
@@ -1,3 +1,30 @@
 # from django.shortcuts import render
+from django.utils.decorators import decorator_from_middleware
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from . import middleware, models, serializers
 
 # Create your views here.
+
+
+@api_view(['GET'])
+def home(_request):
+    """Render the most basic home page ever designed."""
+    return Response('Analytics app seems to be running.')
+
+
+@api_view(['GET', 'POST'])
+@decorator_from_middleware(middleware.AnalyticsTokenAuthMiddleware)
+def event_list(request):
+    """List all events or create a new event."""
+    if request.method == 'GET':
+        events = models.Event.objects.all()
+        serializer = serializers.EventSerializer(events, many=True)
+        return Response(serializer.data)
+    if request.method == 'POST':
+        serializer = serializers.EventSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(created_by=request.user)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
zmieniłbym na HTTP_405 method not allowed ERROR

Reply:
Przejście na generic view załatwiło sprawę, rozszerzyłem test o sprawdzenie tego dla DELETE w f624fd4914842d0ffe8329d7ec24d3a99e5df91e